code/modules/mob/living/carbon/life.dm

```diff
@@ -217,6 +218,35 @@
 	if (breath_gases[/datum/gas/nitryl])
 		var/nitryl_partialpressure = (breath_gases[/datum/gas/nitryl][MOLES]/breath.total_moles())*breath_pressure
 		adjustFireLoss(nitryl_partialpressure/4)
+	//MIASMA
+	if (breath_gases[/datum/gas/miasma])
+		var/miasma_partialpressure = (breath_gases[/datum/gas/miasma][MOLES]/breath.total_moles())*breath_pressure
+
+		if(prob(1 * miasma_partialpressure))
+			var/datum/disease/advance/miasma_disease = new/datum/disease/advance
+			miasma_disease.symptoms = miasma_disease.GenerateSymptoms(1,3)

```

Comment:
`var/datum/disease/advance/miasma_disease = new/datum/disease/advance/random(max_symptoms, 3)` [either keep max_symptoms empty for a random amount or set it yourself]

Reply:
The first 2 vars are needed for min level and max level. Some debug and coderbus symptoms I saw somewhere had level 0 and I don't want to include them.

Reply:
Dont worry, random diseases exist for this exact purpose and can't pick level 0 symptoms